src/core/server/http/http_config.ts

```diff
@@ -153,7 +153,15 @@ export class HttpConfig {
     this.host = rawHttpConfig.host;
     this.port = rawHttpConfig.port;
     this.cors = rawHttpConfig.cors;
-    this.customResponseHeaders = rawHttpConfig.customResponseHeaders;
+    this.customResponseHeaders = Object.entries(rawHttpConfig.customResponseHeaders ?? {}).reduce(
```

Comment:
Lot of tests are using partials of the config. I tried to fix them, but adding `customResponseHeaders: {}` in some was breaking snapshots or even assertions about the received config, so I choose to add this `?? {}` to avoid breaking tests.